---
identifier: "CVE-2021-29570"
package_slug: "maven/org.tensorflow/parentpom"
title: "Out-of-bounds Read"
description: "TensorFlow is an end-to-end open source platform for machine learning.
  The implementation of `tf.raw_ops.MaxPoolGradWithArgmax` can cause reads outside
  of bounds of heap allocated data if attacker supplies specially crafted inputs.
  The implementation(https://github.com/tensorflow/tensorflow/blob/ef0c008ee84bad91ec6725ddc42091e19a30cf0e/tensorflow/core/kernels/maxpooling_op.cc#L1016-L1017)
  uses the same value to index in two different arrays but there is no guarantee that
  the sizes are identical. The fix will be included in TensorFlow We will also cherrypick
  this commit on TensorFlow, TensorFlow, TensorFlow and TensorFlow, as these are also
  affected and still in supported range."
date: "2021-05-18"
pubdate: "2021-05-14"
affected_range: "(,2.1.4),[2.2.0,2.2.3),[2.3.0,2.3.3),[2.4.0,2.4.2)"
fixed_versions: []
affected_versions: "All versions before 2.1.4, all versions starting from 2.2.0 before
  2.2.3, all versions starting from 2.3.0 before 2.3.3, all versions starting from
  2.4.0 before 2.4.2"
not_impacted: ""
solution: "Unfortunately, there is no solution available yet."
urls:
- "https://nvd.nist.gov/vuln/detail/CVE-2021-29570"
cvss_v2: "AV:L/AC:L/Au:N/C:P/I:N/A:P"
cvss_v3: "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:N/A:H"
uuid: "9c6742f8-60ed-41f9-bf5c-c0a31d8ee7c9"
